analyzers/conversion_overflow.go

```diff
@@ -161,6 +147,87 @@ func isStringToIntConversion(instr *ssa.Convert, dstType string) bool {
 	}
 }
 
+func hasExplicitRangeCheck(instr *ssa.Convert, dstType string) bool {
+	block := instr.Block()
+	dstInt, err := parseIntType(dstType)
+	if err != nil {
+		return false
+	}
+
+	minBoundChecked := false
+	maxBoundChecked := false
+
+	srcInt, err := parseIntType(instr.X.Type().String())
+	if err != nil {
+		return false
+	}
+
+	minBoundChecked = checkSourceMinBound(srcInt, dstInt)
+	maxBoundChecked = checkSourceMaxBound(srcInt, dstInt)
+
+	// If both bounds are already checked, return true
+	if minBoundChecked && maxBoundChecked {
+		return true
+	}
+
+	// Recursive function to check predecessors
+	var checkPredecessors func(block *ssa.BasicBlock) bool
+	checkPredecessors = func(block *ssa.BasicBlock) bool {
+		for _, pred := range block.Preds {
+			minChecked, maxChecked := checkBlockForRangeCheck(pred, instr, dstInt)
+			if minChecked {
+				minBoundChecked = true
+			}
+			if maxChecked {
+				maxBoundChecked = true
+			}
+			if minBoundChecked && maxBoundChecked {
+				return true
+			}
+			if checkPredecessors(pred) {
+				return true
+			}
+		}
+		return false
+	}
+
+	// Start checking from the initial block
+	checkPredecessors(block)
+
+	if minBoundChecked && maxBoundChecked {
+		return true
+	}
+
+	return false
+}
+
+func checkBlockForRangeCheck(block *ssa.BasicBlock, instr *ssa.Convert, dstInt integer) (bool, bool) {
+	minBoundChecked := false
+	maxBoundChecked := false
+
+	for _, i := range block.Instrs {
+		if binOp, ok := i.(*ssa.BinOp); ok && isRelevantBinOp(binOp, instr.X) {
+			constVal := extractConst(binOp)
+			if constVal == nil {
+				continue
+			}
+
+			value, err := strconv.ParseInt(constVal.Value.String(), 10, 64)
+			if err != nil {
+				continue
+			}
+
+			minBoundChecked = minBoundChecked || checkMinBoundValue(value, dstInt)
```

Comment:
Doesn't min vs. max need to be determined based on (1) the op, and (2) which side of the op `x` appears on?

Reply:
resolved